File: 04.Simulation模拟/leetcode_796_rotateString.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        if len(s) != len(goal):
            return False
        cur = s
        for i in range(len(s)):
            cur = cur[1:]+cur[0]
            if cur == goal:
                return True
        return False

if __name__ == "__main__":
    sl = Solution()
    s = "abcde"
    goal = "cdeab"
    print(sl.rotateString(s, goal))

File: 04.Simulation模拟/leetcode_796_rotateString.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    bool rotateString(string s, string goal) {
        string cur = s;
        for (int i=0; i<s.size(); i++){
            cur = cur.substr(1)+cur[0];
            if (cur == goal){
                return true;
            }
        }
        return false;
    }
};

int main(){
    Solution sl;
    string s = "abcde", goal = "cdeab";
    cout << sl.rotateString(s, goal) << endl;
}
